homeassistant/components/unifi_direct/device_tracker.py

```diff
@@ -65,17 +65,19 @@ def scan_devices(self):
 
     def get_device_name(self, device):
         """Return the name of the given device or None if we don't know."""
-        hostname = next(
-            (
-                value.get("hostname")
-                for key, value in self.last_results.items()
-                if key.upper() == device.upper()
-            ),
-            None,
-        )
-        if hostname is not None:
-            hostname = str(hostname)
-        return hostname
+        client = self.last_results.get(device.lower())
+        if client:
+            return client.get("hostname") or None
+        return None
+
+    def get_extra_attributes(self, device: str):
+        """Get extra attributes of clients."""
+        client = self.last_results.get(device.lower())
+        extra = {}
+        if client:
+            extra["bssid"] = client["bssid"]
+            extra["ap_name"] = client["ap_name"]
```

Comment:
Shouldn't these be dedicated sensors for each device?

Reply:
Do you mean separate entities? If so how would you suggest to link the clients to that? 

As I said in my description the idea here to add 'location' data to the tracker

Reply:
This is analog with how it's done in openwrt [see](https://github.com/fbradyirl/openwrt-luci-rpc/blob/master/openwrt_luci_rpc/openwrt_luci_rpc.py#L181) and in unifi [see](https://github.com/home-assistant/core/blob/dev/homeassistant/components/unifi/device_tracker.py#L35)